troposphere/rds.py

```diff
@@ -137,7 +137,7 @@ class DBInstance(AWSObject):
         'DBSecurityGroups': (list, False),
         'DBSnapshotIdentifier': (basestring, False),
         'DBSubnetGroupName': (basestring, False),
-        'Engine': (validate_engine, True),
```

Comment:
Good question. The property checks (type or validate functions) are only called if the property is specified in the class instantiation. And then the class validate function is called later on to do validation across all of the properties.